millicore_test.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"errors"
+	"path/filepath"
+	"reflect"
+	"testing"
+)
+
+func TestGetMillicoreConfigTasks(t *testing.T) {
+	tasks := make(chan Task, 1)
+	runner := &FakeRunner{}
+
+	GetMillicoreConfigTasks(tasks, runner, []string{"project1"}, func(project, resource, substr string) ([]string, error) {
+		return []string{"millicore-1"}, nil
+	})
+
+	task := <-tasks
+	err := task()
+	if err != nil {
+		t.Fatal(err)
+	}
+	expectedCalls := []RunCall{
+		{
+			[]string{"oc", "exec", "millicore-1", "--", "cat", "/etc/feedhenry/cluster-override.properties"},
+			filepath.Join("projects", "project1", "millicore", "millicore-1_cluster-override.properties"),
+		},
+	}
+
+	if !reflect.DeepEqual(runner.Calls, expectedCalls) {
+		t.Errorf("runner.Calls = %q, want %q", runner.Calls, expectedCalls)
+	}
+
+}
+
+func TestMillicorePodError(t *testing.T) {
+	tasks := make(chan Task, 1)
+	runner := &FakeRunner{}
+
+	GetMillicoreConfigTasks(tasks, runner, []string{"project1"}, func(project, resource, substr string) ([]string, error) {
+		return nil, errors.New("error retrieving pods")
```

Comment:
The actualy `getResourceNamesBySubstr` can return an error when the oc command returns an error. This test is just testing that when an error is returned from that function it does come through the channel correctly. 

It also covers an otherwise uncovered part of the millicore code.